extensions/wikia/Helios/User.class.php

```diff
@@ -41,4 +41,56 @@ public static function newFromToken( \WebRequest $oRequest )
 		}
 	}
 
+	/**
+	 * Hooked to the UserComparePasswords event, authenticates a user.
+	 *
+	 * Sets &$bResult to false and returns false if the authentication fails.
+	 * Sets &$bResult to false and returns true if the authentication cannot be done.
+	 * Sets &$bResult to true and returns false if the authentication succeeds.
+	 *
+	 * @param string &$sHash string of the password hash (from the database)
+	 * @param string &$sPassword string of the plaintext password the user entered
+	 * @param integer &$iUserId integer of the user's ID or Boolean false if the user ID was not supplied
+	 * @param boolean &$bResult on false returned, this value will be checked to determine if the password was valid
+	 *
+	 * @return boolean false if the authentication has been done, true otherwise (yeah, I know...)
+	 */
+	public static function comparePasswords( &$sHash, &$sPassword, &$iUserId, &$bResult )
+	{
+		// Only proceed for the given percentage of login attempts.
+		global $wgHeliosLoginSamplingRate;
+		if ( rand(0, 100) > $wgHeliosLoginSamplingRate ) {
+			return true;
+		}
+
+		$oLogger = \Wikia\Logger\WikiaLogger::instance();
+		$oLogger->info( 'HELIOS_LOGIN', [ 'method' => __METHOD__ ] );
+		
+		// Get the user's name from the request context.
+		$sUserName= \RequestContext::getMain()->getRequest()->getText( 'username' );
+
+		global $wgHeliosBaseUri, $wgHeliosClientId, $wgHeliosClientSecret;
+		$oHelios = new Client( $wgHeliosBaseUri, $wgHeliosClientId, $wgHeliosClientSecret );
```

Comment:
There's a trait that can be used instead of calling `\Wikia\Logger\WikiaLogger::instance`

```php
namespace Wikia\Logger;
trait Loggable
```